### PR TITLE
HPCC-18061 DOCS:XREF does not display Thor cluster

### PR DESCRIPTION
**Pull request by g-pan**
Fix HPCC-18061 DOCS:XREF does not display Thor cluster
Add documentation for XREF case on a multi-Thor system
where it does not display any or all Thor clusters.
(due to cluster not matching nodeGroup attribute)
Unit test and resultant PDF can be viewed in the ECL Watch manual
found on : http://10.240.32.243/view/Docs/job/DocDev0V/

@JamesDeFabia please review
@jakesmith please review 

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Unit test and resultant PDF can be viewed in the ECL Watch manual
found on : http://10.240.32.243/view/Docs/job/DocDev0V/

### CODE REVIEW

**Comment by g-pan** [comment]
To resolve the conflict, lines 2340 - 2400 should be removed from the conflicted file.

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
For screenshot --ECLWA436.jpg, it makes more sense to check the checkboxes next to both orphan files. You wouldn't delete one file part and not delete the replicated file part.

**Review by JamesDeFabia — APPROVED** [review]
Looks good from my POV. @jakesmith should have the last word

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 1]
```diff
@@ -1133,150 +1130,169 @@
...
 
-        <para>After XRef completes you will see a list of available reports.
-        Click on one of the links to see the results page of each type.</para>
+        <sect3 id="ECLWatchXREFMultiThor">
+          <title>XREF with multiple Thor clusters</title>
 
-        <figure>
-          <title>XRef Errors</title>
+          <para>XREF runs on the primary Thor cluster. For a configuration
+          with more than one Thor cluster, ECL Watch only displays the Thor
```
Comment (jakesmith) [inline docs/ECLWatch/TheECLWatchMan.xml]: >For a configuration with more than one Thor cluster

Need to be clear that it's not any configuration with greater than 1 Thor, but clusters with greater than 1 Thor on the same physical group.

Perhaps:

> For a configuration with more than one Thor cluster on the same physical node group, ...

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 2]
```diff
@@ -1133,150 +1130,169 @@
...
+          nodes listed as the primary Thor (the Thor cluster whose name
+          matches the nodeGroup in the configuration.)</para>
```
Comment (jakesmith) [inline docs/ECLWatch/TheECLWatchMan.xml]: It doesn't display nodes anywhere, does it? It displays file parts.


Should the whole of the above comment in fact be talking about which Thor's are listed, not nodes, or parts.
Eclwatch XREF will only show the primary Thor not the other Thor's that share the same node group.

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 3]
```diff
@@ -1133,150 +1130,169 @@
...
             <mediaobject>
               <imageobject>
-                <imagedata fileref="images/ECLWA437.jpg" vendor="eclwatchSS" />
+                <imagedata fileref="images/ECLWA436.jpg" vendor="eclwatchSS" />
               </imageobject>
             </mediaobject>
-          </figure></para>
+          </figure>
+
+          <para>Typically, these files are deleted. To Delete, check the boxes
```
Comment (jakesmith) [inline docs/ECLWatch/TheECLWatchMan.xml]: >Typically, these files are deleted.

Perhaps:

>Typically, these files will need to be deleted.

**Review by jakesmith — CHANGES_REQUESTED** [review]

**Comment by jakesmith** [comment]
@g-pan - a few mostly minor suggested changes.

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 4]
```diff
@@ -1133,150 +1130,169 @@
...
-      and found" files. These "orphan" are files which are found on the nodes
-      but not registered in the Distributed File System.</para>
+      <sect2 id="The_XREF_Utility">
+        <title>The XRef Utility</title>
 
-      <variablelist>
-        <varlistentry>
-          <term>Found File</term>
+        <para>The XREF utility provides the ability to find "orphaned" and
+        "lost and found" files. These "orphan" are files which are found on
```
[the remark this answers is not in the document]
Comment (JamesDeFabia) [reply docs/ECLWatch/TheECLWatchMan.xml]: our style is to use the Oxford comma, so:
**to find "orphaned", "lost", and "found" files.**

**Comment by g-pan** [comment]
@jakesmith please review changes. 
@richardkchapman I don't understand this conflict, not showing up in my local branch.

**Comment by jakesmith** [comment]
>I don't understand this conflict, not showing up in my local branch.
@g-pan - the commit SHA your commit is based on is not up to date. i.e. it is not the same as the upstream candidate-6.4.x SHA

**Comment by jakesmith** [comment]
@g-pan - apart from rebasing issue, changes look good.

**Comment by g-pan** [comment]
@jakesmith Thank You

**Pull request merged**